packages/pdf-viewer/pdfSource.ts

```diff
@@ -0,0 +1,68 @@
+import * as pdfjsLib from 'pdfjs-dist';
+
+
+export interface ScaledSize {
+	height: number;
+	width: number;
+	scale: number;
+}
+
+export class PdfData {
+	public url: string | Uint8Array;
+	private doc: any = null;
+	public pageCount: number = null;
+	private pages: any = {};
+	private pageSize: {
+		height: number;
+		width: number;
+	} = null;
+	public constructor() {
+	}
+	public loadDoc = async (url: string | Uint8Array) => {
+		this.url = url;
+		const loadingTask = pdfjsLib.getDocument(url);
+		try {
+			const pdfDocument: any = await loadingTask.promise;
+			this.doc = pdfDocument;
+			this.pageCount = pdfDocument.numPages;
+		} catch (err) {
+			throw new Error('Could not load document');
```

Comment:
Please don't swallow exceptions. Use this pattern so that line numbers and original errors are preserved:

err.message = "Could not load document: " + err.message;
throw err.

Also name it "error" not "err".